clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -209,6 +215,34 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
         return ConsumerRecords.empty();
     }
 
+    /**
+     * Set the fetch position to the committed position (if there is one) or reset it using the
+     * offset reset policy the user has configured.
+     *
+     * @return true if the operation completed without timing out
+     * @throws org.apache.kafka.common.errors.AuthenticationException if authentication fails. See the exception for more details
+     * @throws NoOffsetForPartitionException                          If no offset is stored for a given partition and no offset reset policy is
+     *                                                                defined
+     */
+    private boolean updateFetchPositions() {
+        // If any partitions have been truncated due to a leader change, we need to validate the offsets
+        ValidatePositionsApplicationEvent validatePositionsEvent = new ValidatePositionsApplicationEvent();
+        eventHandler.add(validatePositionsEvent);
+
+        // TODO: integrate logic for refreshing committed offsets if available
```

Comment:
Is there a follow up pr for this? or we are still missing the refresh committed offsets

Reply:
That's already implemented (in the integration branch), just included in a different PR that should come after this, following the merge plan.

Reply:
For the record, even though the change for using committed offsets here is 2 lines of code replacing this TODO, it requires also a minor refactoring on existing core components (ex. `ConsumerCoordinator`). So it will all follow this, on a PR like "Adding support for using committed offsets when updating fetch positions"